pkg/operator2/workload/sync_openshift_oauth_apiserver.go

```diff
@@ -0,0 +1,286 @@
+		required.Annotations = map[string]string{}
+	}
+	required.Annotations["openshiftapiservers.operator.openshift.io/pull-spec"] = c.targetImagePullSpec
+	required.Annotations["openshiftapiservers.operator.openshift.io/operator-pull-spec"] = c.operatorImagePullSpec
+
+	required.Labels["revision"] = strconv.Itoa(int(authOperator.Status.OAuthAPIServer.LatestAvailableRevision))
+	required.Spec.Template.Labels["revision"] = strconv.Itoa(int(authOperator.Status.OAuthAPIServer.LatestAvailableRevision))
+
+	// we watch some resources so that our deployment will redeploy without explicitly and carefully ordered resource creation
+	inputHashes, err := resourcehash.MultipleObjectHashStringMapForObjectReferences(
+		c.kubeClient,
+		resourcehash.NewObjectRef().ForConfigMap().InNamespace(c.targetNamespace).Named("config"),
+		resourcehash.NewObjectRef().ForSecret().InNamespace(c.targetNamespace).Named("etcd-client"),
+		resourcehash.NewObjectRef().ForConfigMap().InNamespace(c.targetNamespace).Named("etcd-serving-ca"),
+		resourcehash.NewObjectRef().ForConfigMap().InNamespace(c.targetNamespace).Named("trusted-ca-bundle"),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("invalid dependency reference: %q", err)
+	}
+
+	for k, v := range inputHashes {
+		annotationKey := fmt.Sprintf("operator.openshift.io/dep-%s", k)
+		required.Annotations[annotationKey] = v
+		if required.Spec.Template.Annotations == nil {
+			required.Spec.Template.Annotations = map[string]string{}
+		}
+		required.Spec.Template.Annotations[annotationKey] = v
+	}
+
+	err = c.ensureAtMostOnePodPerNode(&required.Spec)
+	if err != nil {
+		return nil, fmt.Errorf("unable to ensure at most one pod per node: %v", err)
+	}
+
+	// Set the replica count to the number of master nodes.
+	masterNodeCount, err := c.countNodes(required.Spec.Template.Spec.NodeSelector)
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine number of master nodes: %v", err)
+	}
+	required.Spec.Replicas = masterNodeCount
+	// Set the replica count as an annotation to ensure that ApplyDeployment
+	// will update the deployment in the API when the replica count
+	// changes. Updates are otherwise skipped if the metadata matches and the
+	// generation is up-to-date.
+	required.Annotations["openshiftapiservers.operator.openshift.io/replicas"] = fmt.Sprintf("%d", *masterNodeCount)
+
+	deployment, _, err := resourceapply.ApplyDeployment(c.kubeClient.AppsV1(), c.eventRecorder, required, resourcemerge.ExpectedDeploymentGeneration(required, generationStatus), false)
+	return deployment, err
+}
+
+// oAuthAPIServerConfig hold configuration for this controller it's taken from ObservedConfig.Raw
+// note that this struct is unsupported in a sense that it's not exposed through API
+type oAuthAPIServerConfig struct {
+	APIServerArguments map[string][]string `json:"apiServerArguments"`
+}
+
+// unstructuredConfigFrom extract raw config for this controller
+func unstructuredConfigFrom(rawCfg []byte) ([]byte, error) {
```

Comment:
feel free to export and re-use https://github.com/openshift/cluster-authentication-operator/blob/79b4c66c56284f2424b181de4220b52093c0e78c/pkg/operator2/oauth.go#L153-L171